src/js/hyperschedule.ts

```diff
@@ -527,9 +527,51 @@ function coursePassesTextFilters(
   return true;
 }
 
-function parseDaysInequality(inputDays: string) {
-  for (const rel of filterInequalities)
-    if (inputDays.startsWith(rel)) return rel;
+// timeFilters is a one or two element array
+// [start_time] or [start_time, end_time]
+function coursePassesTimeFilters(
+  course: Course.CourseV3,
+  timeFilters: string[]
+) {
+  // indicates no current time filters
+  if (timeFilters[0] === "") {
+    return true;
+  }
+
+  const startTime = parseFloat(timeFilters[0].replace(":", "."));
```

Comment:
Eh?  This seems kind of like a strange way to parse times--consider the following inputs:
- `5:3`--should this be either considered invalid, or it should parse as `5:03`, but currently it would be equivalent to `5:30`.
- `4:45pm`--`parseFloat` will simply ignore the `pm`, contradicting the example usage provided by the "help" popup.  Or am I overlooking some other code that does some pre-processing here?

Also, maybe all this new filtering code should be placed in a separate, new, `filter.ts` module, so as to avoid cluttering up `hyperschedule.ts` even more.  I'm in a slow ongoing process to split up `hyperschedule.ts` into smaller modules; filters are a good place to start.

Reply:
Another potential issue with using `parseFloat` is that floating-points are more susceptible to rounding errors, possibly, which may lead to, _maybe_, in some cases, `9:35` becoming `9.35000001` or something and not matching a `9:35` class.  It doesn't seem very likely, but maybe it's good to avoid it if we can, anyway.

Reply:
The elements within the time filters are already processed through `getTimeFilter` so that means all elements are between 00:00 and 23:59, without "am"/"pm". I believe we used `parseFloat` for the second case where there could be a range of times and thus checking if a course's start time is between the range of times is easier with floats. For the first case, string comparison would be enough. However, the second case is trickier with the range. There is a way with `parseInt` instead, where we can convert the times to number of minutes and then compare that. Does that sound like a good way to you (gets past `parseFloat` at least)?